src/constants/parseJson.ts

```diff
@@ -0,0 +1,45 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// copied from vscode-azurefunctions extension
+
+import * as jsonc from 'jsonc-parser';
+import * as constants from './constants';
+
+/**
+ * Parses and returns JSON
+ * Has extra logic to remove a BOM character if it exists and handle comments
+ */
+export function parseJson<T extends object>(data: string): T {
+	if (data.charCodeAt(0) === 0xFEFF) {
+		data = data.slice(1);
+	}
+
+	const errors: jsonc.ParseError[] = [];
+	const result: T = <T>jsonc.parse(data, errors, { allowTrailingComma: true });
```

Comment:
Is this from something? Is there a reason you can't just use JSON.parse instead of relying on a 3rd party package?

Reply:
This is used in getLocalSettingsJson that was reused from the sql-database-projects repo.